app/controllers/declaration/PreviousDocumentsController.scala

```diff
@@ -19,79 +19,55 @@ package controllers.declaration
   }
 
   def savePreviousDocuments(mode: Mode): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    import MultipleItemsHelper._
-
-    val boundForm = form().bindFromRequest()
-    val actionTypeOpt = FormAction.bindFromRequest()
+    val boundForm = if (isFirstPreviousDocument) Document.treatLikeOptional(form().bindFromRequest()) else form().bindFromRequest()
 
     val cache = request.cacheModel.previousDocuments.getOrElse(PreviousDocumentsData(Seq.empty))
 
-    actionTypeOpt match {
-      case SaveAndContinue | SaveAndReturn =>
-        saveAndContinue(boundForm, cache.documents, isScreenMandatory, maxAmountOfItems).fold(
-          formWithErrors => Future.successful(BadRequest(previousDocumentsPage(mode, formWithErrors, cache.documents))),
-          updatedCache =>
-            updateCache(PreviousDocumentsData(updatedCache))
-              .map(_ => navigator.continueTo(mode, controllers.declaration.routes.ItemsSummaryController.displayPage))
-        )
-
-      case Add =>
-        add(boundForm, cache.documents, PreviousDocumentsData.maxAmountOfItems).fold(
-          formWithErrors => Future.successful(BadRequest(previousDocumentsPage(mode, formWithErrors, cache.documents))),
+    if (boundForm.value.isEmpty && boundForm.errors.isEmpty)
+      updateCache(PreviousDocumentsData(Seq.empty)).map { _ =>
+        navigator.continueTo(mode, controllers.declaration.routes.ItemsSummaryController.displayPage)
+      } else
+      MultipleItemsHelper
+        .add(boundForm, cache.documents, PreviousDocumentsData.maxAmountOfItems)
+        .fold(
+          formWithErrors => Future.successful(BadRequest(previousDocumentsPage(mode, formWithErrors))),
           updatedCache =>
             updateCache(PreviousDocumentsData(updatedCache))
-              .map(_ => navigator.continueTo(mode, controllers.declaration.routes.PreviousDocumentsController.displayPage))
+              .map(_ => navigator.continueTo(mode, controllers.declaration.routes.PreviousDocumentsSummaryController.displayPage))
         )
-
-      case Remove(ids) =>
-        val itemToRemove = Document.fromJsonString(ids.head)
-        val updatedDocuments = MultipleItemsHelper.remove(cache.documents, itemToRemove.contains(_: Document))
-        updateCache(PreviousDocumentsData(updatedDocuments))
-          .map(_ => navigator.continueTo(mode, routes.PreviousDocumentsController.displayPage))
-
-      case _ => Future.successful(BadRequest(previousDocumentsPage(mode, boundForm, cache.documents)))
-    }
   }
 
   private def updateCache(formData: PreviousDocumentsData)(implicit req: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
     updateExportsDeclarationSyncDirect(model => model.copy(previousDocuments = Some(formData)))
+
+  private def isFirstPreviousDocument()(implicit request: JourneyRequest[AnyContent]): Boolean =
+    request.cacheModel.previousDocuments.getOrElse(PreviousDocumentsData(Seq.empty)).documents.isEmpty
```

Comment:
Instead of `getOrElse` and then `isEmpty`, could it be replaced with `exists` ?

Reply:
Good point, always forgot about this, thanks, will do it

Reply:
Fixed